staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/admission.go

```diff
@@ -82,21 +82,26 @@ type liveLookupEntry struct {
 // NewResourceQuota configures an admission controller that can enforce quota constraints
 // using the provided registry.  The registry must have the capability to handle group/kinds that
 // are persisted by the server this admission controller is intercepting
-func NewResourceQuota(config *resourcequotaapi.Configuration, numEvaluators int, stopCh <-chan struct{}) (*QuotaAdmission, error) {
+func NewResourceQuota(config *resourcequotaapi.Configuration, numEvaluators int) (*QuotaAdmission, error) {
 	quotaAccessor, err := newQuotaAccessor()
 	if err != nil {
 		return nil, err
 	}
 
 	return &QuotaAdmission{
 		Handler:       admission.NewHandler(admission.Create, admission.Update),
-		stopCh:        stopCh,
+		stopCh:        nil,
```

Comment:
it's hard to reason about this being nil at init time and set to a non-nil thing later ... how do we prevent things from propagating handles to this further down into the plugin/controller/etc and not picking up the update in SetShutdownNotification?

Reply:
We don't have a generic mechanism for preventing propagating things down the stack before stopCh. But we don't even have a mechanism for setting stopCh now, so it's strict improvement.

I think it's mostly the same as for any regular bug - we need to catch that during the review.
In this PR I'm explicitly setting "initialization error" if we try to initialize something that requires stopCh and stopCh is unset, which will ensure that plugin will later not pass the validation.